samples/extensions/vertex_dynamic_state/vertex_dynamic_state.cpp

```diff
@@ -350,19 +350,37 @@ void VertexDynamicState::build_command_buffers()
 		vkCmdBindDescriptorSets(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &descriptor_set, 0, nullptr);
 
 		/* skybox */
-		change_vertex_input_data(VERTEX_DYNAMIC_STATE_USE_FRAMEWORK_VERTEX_STRUCT);
-		vkCmdSetVertexInputEXT(draw_cmd_buffer, 1, &vertex_bindings_description_ext, 2, vertex_attribute_description_ext);
+		/* First set of vertex input dynamic data (Vertex structure) */
+		vertex_bindings_description_ext[0].stride  = sizeof(Vertex);
+		vertex_attribute_description_ext[1].offset = sizeof(float) * 3;
```

Comment:
Maybe even better: `offsetof(Vertex, normal);` ?
And `offsetof(SampleVertex, normal)` for the other case?

Reply:
I tried using `offsetof` before (see commit: https://github.com/KhronosGroup/Vulkan-Samples/commit/73330043bcc11d52fc012f0dae096601ea600336).
Unfortunately, I have CI failures under android and MacOS because of using `offsetof`, and I just back to `sizeof` method.

Reply:
That's pretty strange! What errors did you get there?
Did you `#include <cstddef>`?

Reply:
Unfortunately, as I see logs from previous commit CI are overwritten by a new CI check.
I didn't include `<cstddef>`, but I do mini research and this is included by going: <br> `api_vulkan_sample.h` -> `gui.h` -> `platform/input_events.h` and in last header we have `#include <cstddef>`.
As far as I know, because of this include daisy-chain it is not necessary to include `<cstddef>` directly in my sample header.

Reply:
Indeed, you should not need to include `cstddef` then.
But `offsetof` is used in other places in this repository, obviously without any issues. So maybe you could give it another try?

Reply:
Sure, probably I know what I have done not quite right. I will commit changes within 30 minutes (please run CI check after that) :)

Reply:
Okay, previously I use maybe strange expressions and Android and MacOS hate it, but now it works fine.

Reply:
Of course, I just want to focus on refactoring code in the first place. From your side, do you have plans to give me more suggestions? Or can I assume that's close to the end? :)

Reply:
Nope, I don't see any additional issues.
Just the two yet to be resolved on getting the function pointer, and maybe some words on performance.

Reply:
Okay, Thanks for your work and time. I will try to resolve them.